source/uwp/Renderer/lib/AdaptiveActionRendererRegistration.h

```diff
@@ -2,29 +2,28 @@
 // Licensed under the MIT License.
 #pragma once
 
-#include "AdaptiveCards.Rendering.Uwp.h"
-#include "Util.h"
+#include "AdaptiveActionRendererRegistration.g.h"
 
-namespace AdaptiveCards::Rendering::Uwp
+namespace winrt::AdaptiveCards::Rendering::Uwp::implementation
 {
-    class AdaptiveActionRendererRegistration
-        : public Microsoft::WRL::RuntimeClass<Microsoft::WRL::RuntimeClassFlags<Microsoft::WRL::RuntimeClassType::WinRtClassicComMix>,
-                                              Microsoft::WRL::Implements<ABI::AdaptiveCards::Rendering::Uwp::IAdaptiveActionRendererRegistration>,
-                                              Microsoft::WRL::FtmBase>
+    struct AdaptiveActionRendererRegistration : AdaptiveActionRendererRegistrationT<AdaptiveActionRendererRegistration>
     {
-        AdaptiveRuntime(AdaptiveActionRendererRegistration);
+        using RegistrationMap =
+            std::unordered_map<winrt::hstring, winrt::IAdaptiveActionRenderer, ::AdaptiveCards::CaseInsensitiveHash, ::AdaptiveCards::CaseInsensitiveEqualTo>;
```

Comment:
>unordered_map [](http://example.com/codeflow?start=17&length=13)

I believe std::map<winrt::hstring, winrt::IAdaptiveActionRenderer> would be enough. we don't need unordered_map.

Reply:
**std::unordered_map** seems to be more efficient than usual map in retrieving the objects [source](https://stackoverflow.com/a/3999891)

Reply:
For small amount of data, there is no big difference between unordered_map and map. but unordered_map is more complex because you have to implement hash and compare function here.

Reply:
@paulcam206 do you know why custom hash function and compare methods were implemented?